aws/scripts/common.sh

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env bash
+
+# Copyright 2021 DataStax, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# Common commands for all scripts              
+
+# Locate the root directory. Used by scripts that source this one.
+# shellcheck disable=SC2034
+ROOT="$( cd "$( dirname "${BASH_SOURCE[0]}" )/.." && pwd )"
+
+# git is required for this tutorial
+# https://git-scm.com/book/en/v2/Getting-Started-Installing-Git
+command -v git >/dev/null 2>&1 || { \
+ echo >&2 "I require git but it's not installed.  Aborting."
+ echo >&2 "Refer to: https://git-scm.com/book/en/v2/Getting-Started-Installing-Git"
+ exit 1
+}
+
+# glcoud is required for this tutorial
+# https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2-linux.html
+command -v aws >/dev/null 2>&1 || { \
+ echo >&2 "I require aws cliv2 but it's not installed.  Aborting."
+ echo >&2 "Refer to: https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2-linux.html"
+ exit 1
+}
+
+# Make sure kubectl is installed.  If not, refer to:
+# https://kubernetes.io/docs/tasks/tools/install-kubectl/
+command -v kubectl >/dev/null 2>&1 || { \
+ echo >&2 "I require kubectl but it's not installed.  Aborting."
+ echo >&2 "Refer to: https://kubernetes.io/docs/tasks/tools/install-kubectl/"
+ exit 1
+}
+
+# Make sure Helm is installed. If not, refer to:
+# https://helm.sh/docs/intro/install/
+command -v helm >/dev/null 2>&1 || { \
+ echo >&2 "I require helm but it's not installed.  Aborting."
+ echo >&2 "Refer to: https://helm.sh/docs/intro/install/"
+ exit 1
+}
+
+# Make sure Terraform0.14 is installed. If not, refer to:
+# https://www.terraform.io/docs/cli/install/apt.html
+command -v terraform >/dev/null 2>&1 || { \
+ echo >&2 "I require terraform 0.14 but it's not installed.  Aborting."
+ echo >&2 "https://www.terraform.io/docs/cli/install/apt.html"
+ echo >&2 "Refer to: sudo apt install terraform=0.14.0"
+ exit 1
+}
+
+# Make sure python3 is installed. If not, refer to:
+# https://www.python.org/downloads/
+command -v python3 >/dev/null 2>&1 || { \
```

Comment:
I'm a little concerned that we're looking for `python3` as a valid command. The alternative is to check `python` and `python3` and look at version numbers which seems like a bit too much. Maybe ignore this comment.